packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -308,6 +308,19 @@ class ToolbarOptions {
 /// movement. This widget does not provide any focus management (e.g.,
 /// tap-to-focus).
 ///
+/// ## Handling User Input
+///
+/// Currently the user may change the text this widget contains via keyboard or
```

Comment:
The user also can change the text by the `controller` directly. If change the text by `controller` [inpitFormatters] will not be applied. Should [inpitFormatters] work here?

Reply:
The "user" here refers to end-users (that use the app). I think `inputFormatters` is meant for sanitizing user input. If the developer decides to change the text programmatically then they should be responsible for making sure the new text is valid.

Reply:
Oh in the documentation it only says "when the text input changes": https://api.flutter.dev/flutter/material/TextField/inputFormatters.html. I think we should make that clear.